soroban-env-host/src/e2e_invoke.rs

```diff
@@ -200,6 +200,35 @@ pub fn get_ledger_changes(
     Ok(changes)
 }
 
+/// Creates ledger changes for entries that don't exist in the storage.
+///
+/// In recording mode it's possible to have discrepancies between the storage
+/// and the footprint. Specifically, if an entry is only accessed from a
+/// function that has failed and had its failure handled gracefully (via
+/// `try_call`), then the storage map will get rolled back and the access will
+/// only be recorded in the footprint. However, we still need to account for
+/// these in the ledger entry changes, as downstream consumers (simulation) rely
+/// on that to determine the fees.
+#[cfg(any(test, feature = "recording_mode"))]
+fn add_footprint_only_ledger_changes(
+    budget: &Budget,
+    storage: &Storage,
+    changes: &mut Vec<LedgerEntryChange>,
+) -> Result<(), HostError> {
+    for (key, access_type) in storage.footprint.0.iter(budget)? {
+        // We have to check if the entry exists in the internal storage map
+        // because `has` check on storage affects the footprint.
+        if storage.map.contains_key(key, budget)? {
+            continue;
+        }
+        let mut entry_change = LedgerEntryChange::default();
+        metered_write_xdr(budget, key.as_ref(), &mut entry_change.encoded_key)?;
+        entry_change.read_only = matches!(*access_type, AccessType::ReadOnly);
```

Comment:
Maybe I'm misunderstanding this issue, but if the failed contract call is writing this data instead of reading, wouldn't the entry need to be in the readWrite set?

Reply:
This is a bool flag that is set based on the recorded footprint access type. So RO entry changes should have `true` and RW entries should have `false`.

Reply:
**facepalm** Sorry I misread this...